app/notification_email_test.go

```diff
@@ -825,3 +825,99 @@ func TestLandingLinkPermalink(t *testing.T) {
 }
+
+func TestMarkdownConversion(t *testing.T) {
+	tests := []struct {
+		name string
+		args string
+		want string
+	}{
+		{
+			name: "markdown: strong",
+			args: "This is **Mattermost**",
+			want: "This is <strong>Mattermost</strong>",
+		},
+		{
+			name: "markdown: emphasis",
+			args: "This is *Mattermost*",
+			want: "This is <em>Mattermost</em>",
+		},
+		{
+			name: "markdown: links",
+			args: "This is [Mattermost](https://mattermost.com)",
+			want: "This is <a href=\"https://mattermost.com\">Mattermost</a>",
+		},
+		{
+			name: "markdown: strikethrough",
+			args: "This is ~~Mattermost~~",
+			want: "This is <del>Mattermost</del>",
+		},
+		{
+			name: "markdown: table",
+			args: "| Tables        | Are           | Cool  |\n" +
+				"| ------------- |:-------------:| -----:|\n" +
+				"| col 3 is      | right-aligned | $1600 |\n" +
+				"| col 2 is      | centered      |   $12 |\n" +
+				"| zebra stripes | are neat      |    $1 |\n",
+			want: "<table>\n" +
+				"<thead>\n" +
+				"<tr>\n" +
+				"<th>Tables</th>\n" +
+				"<th style=\"text-align:center\">Are</th>\n" +
+				"<th style=\"text-align:right\">Cool</th>\n" +
+				"</tr>\n" +
+				"</thead>\n" +
+				"<tbody>\n" +
+				"<tr>\n" +
+				"<td>col 3 is</td>\n" +
+				"<td style=\"text-align:center\">right-aligned</td>\n" +
+				"<td style=\"text-align:right\">$1600</td>\n" +
+				"</tr>\n" +
+				"<tr>\n" +
+				"<td>col 2 is</td>\n" +
+				"<td style=\"text-align:center\">centered</td>\n" +
+				"<td style=\"text-align:right\">$12</td>\n" +
+				"</tr>\n" +
+				"<tr>\n" +
+				"<td>zebra stripes</td>\n" +
+				"<td style=\"text-align:center\">are neat</td>\n" +
+				"<td style=\"text-align:right\">$1</td>\n" +
+				"</tr>\n" +
+				"</tbody>\n" +
+				"</table>",
+		},
+		{
+			name: "markdown: multiline with header and links",
+			args: "###### H6 header\n[link 1](https://mattermost.com) - [link 2](https://mattermost.com)",
+			want: "<h6>H6 header</h6>\n" +
+				"<p><a href=\"https://mattermost.com\">link 1</a> - <a href=\"https://mattermost.com\">link 2</a></p>",
+		},
+	}
+
+	th := SetupWithStoreMock(t)
+	defer th.TearDown()
+
+	recipient := &model.User{}
+	storeMock := th.App.Srv().Store.(*mocks.Store)
+	teamStoreMock := mocks.TeamStore{}
+	teamStoreMock.On("GetByName", "testteam").Return(&model.Team{Name: "testteam"}, nil)
+	storeMock.On("Team").Return(&teamStoreMock)
+	channel := &model.Channel{
+		DisplayName: "ChannelName",
+		Type:        model.ChannelTypeOpen,
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			post := &model.Post{
+				Id:      "Test_id",
+				Message: tt.args,
+			}
+			got, err := th.App.getNotificationEmailBody(recipient, post, channel, "ChannelName", "sender", "testteam", "http://localhost:8065/landing#/testteam", model.EmailNotificationContentsFull, true, i18n.GetUserTranslations("en"), "user-avatar.png")
+			if err != nil {
+				t.Fatalf("error: %v", err)
+			}
```

Comment:
Change this to `require.NoError(t, err)`

Reply:
haha damn! I did have this but for some reason went with the t.Fatalf.